app.js

```diff
@@ -22,9 +22,27 @@ app.use(logger('dev'));
 app.use(express.json({limit: '100mb'}));
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
-app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
+// express-session
+const sess = session({
+    store: new MemoryStore({
+        ttl: 45 * 60 * 1000, // 45 minutes
+        checkPeriod: 30 * 60 * 1000, // half an hour
```

Comment:
Is the check period supposed to be less than the TTL? The docs says `checkPeriod` defines how long MemoryStore will check for expired- I'm assuming this means it'll check in 30 mins intervals? Not sure if I misinterpreted it 😅

Reply:
Does this also mean that we'll automatically delete their session information after 45 minutes? And they'll have to sign back in again? Also just curious opening two different tabs would be two different sessions?

Reply:
Do we also set the secret ourselves?

Reply:
> Is the check period supposed to be less than the TTL?

Yeah I'm actually not too sure about `checkPeriod` lemme go through it. I'm not sure if it means it checks every half hour, or it checks for half an hour.

> Does this also mean that we'll automatically delete their session information after 45 minutes? And they'll have to sign back in again?

Yup!

> Do we also set the secret ourselves?
Yup! I used a random secret, think of it as a random password. This doc explains it pre well https://github.com/expressjs/session#secret

Reply:
Gotcha! How did you get a random secret?

Reply:
Ah wait yes for `checkPeriod` they made a comment `// prune expired entries every 24h`. So to answer your question, every 30 minutes express-session will delete any expired entries. And each entry expires after 45 minutes. 

however, the `ttl` field behaves differently. E.g. I login, we set the token in session to expire after 45 minutes. If I do another request that modifies the session, we reset the 45 minute count. So technically, the cache maybe could never expire. But we would be sure that if the access token is invalid, we simply set it again

Reply:
> Gotcha! How did you get a random secret?

Oh I searched up random password generator and generated smthg like that

Reply:
Oooh, I see!! In terms of another request, thats like any REST request to our app right?

Reply:
Yup!